edalize/symbiflow.py

```diff
@@ -52,31 +59,138 @@ def get_doc(cls, api_ver):
 
             return {
-                "description": "The Symbiflow backend executes Yosys sythesis tool and VPR place and route. It can target multiple different FPGA vendors",
+                "description": "The Symbiflow backend executes Yosys sythesis tool and VPR/Nextpnr place and route. It can target multiple different FPGA vendors",
                 "members": symbiflow_members,
             }
 
     def get_version(self):
         return "1.0"
 
+    def configure_nextpnr(self):
+        (src_files, incdirs) = self._get_fileset_files(force_slash=True)
+        vendor = self.tool_options.get("vendor")
+
+        # Yosys configuration
+        yosys_synth_options = self.tool_options.get("yosys_synth_options", "")
+        yosys_template = self.tool_options.get("yosys_template")
+        yosys_edam = {
+                "files"         : self.files,
+                "name"          : self.name,
+                "toplevel"      : self.toplevel,
+                "parameters"    : self.parameters,
+                "tool_options"  : {
+                                    "yosys" : {
+                                        "arch" : vendor,
+                                        "yosys_synth_options" : yosys_synth_options,
+                                        "yosys_template" : yosys_template,
+                                        "yosys_as_subtool" : True,
+                                    }
+                                }
+                }
+
+        yosys = getattr(import_module("edalize.yosys"), "Yosys")(yosys_edam, self.work_root)
+        yosys.configure()
+
+        # Nextpnr configuration
+        arch = self.tool_options.get("arch")
+        if arch not in getattr(self, "archs"):
+            logger.error('Missing or invalid "arch" parameter: {} in "tool_options"'.format(arch))
+
+        package = self.tool_options.get("package")
+        if not package:
+            logger.error('Missing required "package" parameter')
+
+        schema_dir = os.getenv("INTERCHANGE_SCHEMA_PATH")
+        if not schema_dir and arch is "fpga_interchange":
+            raise RuntimeError("Environment variable INTERCHANGE_SCHEMA_PATH was not found. It should be set to /path/to/fpga-interchange-schema/interchange directory for fpga_interchange variant.")
+
+        part = self.tool_options.get("part")
+        if not part:
+            logger.error('Missing required "part" parameter')
+
+        target_family = None
+        for family in getattr(self, "fpga_interchange_families"):
+            if family in part:
+                target_family = family
+                break
+
+        if target_family is None and arch is "fpga_interchange":
+            logger.error("Couldn't find family for part: {}. Available families: {}".format(part, ", ".join(getattr(self, "fpga_interchange_families"))))
+
+        chipdb = None
+        device = None
+        constr = None
+
+        for f in src_files:
```

Comment:
This will always pick the latest file. At least for xdc files it's not uncommon to have several files. If that's not supported by the backend we should at least warn users that only one of them will be used